### ISSUE TITLE
NotAuthorized API Error??

### ISSUE THREAD

**Issue by KatttTheFemboi** [iBicha/playlet#511]
Reason:NotAuthorized
StatusCode: 401
RequestId: 122683405
Body: API disabled

What is this and what does it mean? I'm assuming it has to be a playlet issue since this happens on all instances I think, but I might be wrong

**Comment by Artemis6428** [comment]
Some servers, at least inv.nadeko, recently disabled their api for some things because youtube has been blocking backends when there's a lot of traffic and bot/api devs are generating too much for them to stay functional.

[Source](https://git.nadeko.net/Fijxu/invidious/wiki/Important-information)

**Comment by Pridefulsubaru** [comment]
Made this account just to say mine is doing the same thing but what’s frustrating I have friends that use it get the same error code hit ok and can use the app perfectly fine it won’t let me do anything. Hopefully they get it worked out.

**Comment by iBicha** [comment]
Yup. I've tagged this as a bug to avoid confusion, but this is not a bug with Playlet, nor with Invidious.
There are simply currently no public instances with API access left.
This is the course of events: as less and less instances become available, the strain on those servers go up. Eventually some maintainers will choose to disable API access to keep the instance working.

**The options right now:**
- How your own instance: https://docs.invidious.io/installation
- Use the YouTube mobile app (not the web app) to cast to Playlet: this part does not rely on Invidious, and is working fine to watch videos (set the backend to "Playlet" in the settings)
- I'm working to make Playlet work independently from Invidious by Implementing a built-in backend #502 and it's going to take a while.

**Comment by Artemis6428** [comment]
At least for now the [jing.rocks](https://invidious.jing.rocks) instance is working for me.

**Comment by iBicha** [comment]
A very basic built-in backend has been released in [Canary](https://github.com/iBicha/playlet/releases/tag/canary)
Many features are missing in this built-in backend (A LOT is missing) but you have the very basic functionality: search, open a channel, open a playlist, play a video.
To use this functionality, simply delete the invidious instance you're using (leave blank) and save.

**Comment by Artemis6428** [comment]
Already using it 😊 Always use the canary and keep up with all commits and PR progress! 🫡

**Comment by iBicha** [comment]
Thanks everyone for your patience. The built-in backend is now released.
It's still missing many features (which I will build up as I go) but it is somewhat usable.
You can still use Invidious instances as usual, that part didn't change.
Closing the issue, since it is about a specific Invidious instance, and not particularly an issue with Playlet
